### ISSUE TITLE
Can’t open onion page of taz.de – Onion-Location redirect too greedy

### ISSUE THREAD

**Issue by glomeg132** [OnionBrowser/iOS-OnionBrowser#382]
I can’t open the onion page of German newspaper taz.de. The address is http://zervmwoc5flabhlh3heegnspbpoebbgr3kgkaarkpnz2gtsuunlxgqyd.onion/

Opening the onion page shows the source code of the page. It does work with Tor on Android.

**Comment by tladesignz** [comment]
Very interesting. Thanks for bringing this to our attention!

TL;DR: Disable "Follow Onion-Location Header automatically" in Settings -> Default Security, to stop this behaviour.

The root cause is, that the taz uses Piwik aka. Matomo. The main HTML page contains JavaScript which initializes Matomo to the hard-coded `taz.de` domain and tries to load a script from there.

That script has an `Onion-Location` header, which Onion Browser redirects to, automatically. Unfortunately, it seems that redirect feature's implementation is broken for this case. It should not load a resource as the main document after the redirect.

2 actions:

1. Improve the `Onion-Location redirect` function.
2. Inform taz.de, that they could improve their setup.

**Comment by tladesignz** [comment]
> TL;DR: Disable "Follow Onion-Location Header automatically" in Settings -> Default Security, to stop this behaviour.

You could also set up a "Custom Site Security" entry for `taz.de` where you disable "Follow Onion-Location Header automatically" to limit this workaround for just this site.

I already fixed the behaviour in `Onion Browser`: In the future, it will only redirect, if the header is served with the main document. Next release probably in a few weeks!

**Comment by tladesignz** [comment]
Just informed taz.de about this.
